### ISSUE TITLE
Unable to run application inside kubernernetes

### ISSUE THREAD

**Issue by kameshsampath** [quarkusio/quarkus#3093]
**Describe the bug**
Unable to start Quarkus application inside kubernetes. 

**Expected behavior**
Application should start correctly 

**Actual behavior**
```lingua-greeter-667949c798-mrvql lingua-greeter java.lang.IllegalStateException: Failed to create cache dir
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.file.impl.FileResolver.setupCacheDir(FileResolver.java:332)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.file.impl.FileResolver.<init>(FileResolver.java:87)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxImpl.<init>(VertxImpl.java:167)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxImpl.vertx(VertxImpl.java:92)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxFactoryImpl.vertx(VertxFactoryImpl.java:34)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.Vertx.vertx(Vertx.java:85)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.vertx.runtime.VertxTemplate.initialize(VertxTemplate.java:102)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.vertx.runtime.VertxTemplate.configureVertx(VertxTemplate.java:41)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.deployment.steps.VertxProcessor$build14.deploy_0(VertxProcessor$build14.zig:120)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.deployment.steps.VertxProcessor$build14.deploy(VertxProcessor$build14.zig:36)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:82)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runtime.Application.start(Application.java:84)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runtime.Application.run(Application.java:196)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:34)
lingua-greeter-667949c798-mrvql lingua-greeter 2019-07-04 16:57:42,307 INFO  [io.sma.rea.mes.ext.MediatorManager] (main) Cancel subscriptions
lingua-greeter-667949c798-mrvql lingua-greeter Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:118)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runtime.Application.start(Application.java:84)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runtime.Application.run(Application.java:196)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:34)
lingua-greeter-667949c798-mrvql lingua-greeter Caused by: java.lang.IllegalStateException: Failed to create cache dir
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.file.impl.FileResolver.setupCacheDir(FileResolver.java:332)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.file.impl.FileResolver.<init>(FileResolver.java:87)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxImpl.<init>(VertxImpl.java:167)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxImpl.vertx(VertxImpl.java:92)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.impl.VertxFactoryImpl.vertx(VertxFactoryImpl.java:34)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.vertx.core.Vertx.vertx(Vertx.java:85)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.vertx.runtime.VertxTemplate.initialize(VertxTemplate.java:102)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.vertx.runtime.VertxTemplate.configureVertx(VertxTemplate.java:41)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.deployment.steps.VertxProcessor$build14.deploy_0(VertxProcessor$build14.zig:120)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.deployment.steps.VertxProcessor$build14.deploy(VertxProcessor$build14.zig:36)
lingua-greeter-667949c798-mrvql lingua-greeter  at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:82)
lingua-greeter-667949c798-mrvql lingua-greeter  ... 3 more
- lingua-greeter-667949c798-mrvql```


**To Reproduce**
Steps to reproduce the behavior:
1. deploy application from http://github.com/kameshsampath/lingua-greeter

```yaml
---
apiVersion: v1
kind: Service
metadata:
  name: lingua-greeter
spec:
  selector:
    app: lingua-greeter
  ports:
  - port: 8080
    targetPort: 8080
  type: NodePort
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: lingua-greeter
spec:
  selector:
    matchLabels:
      app: lingua-greeter
  template:
    metadata:
      labels:
        app: lingua-greeter
    spec:
      containers:
      - name: lingua-greeter
        image: rhdevelopers/lingua-greeter
        imagePullPolicy: IfNotPresent
        env:
         - name: GOOGLE_APPLICATION_CREDENTIALS
           value: /tmp/google-cloud-credentials.json    
         - name: KAFKA_BOOTSTRAP_SERVERS
           value: my-cluster-kafka-bootstrap.kafka:9092    
        resources:
          limits:
            memory: "128Mi"
            cpu: "500m"
        ports:
        - containerPort: 8080
        volumeMounts:
        - name: google-cloud-creds
          mountPath: /tmp
          readOnly: true
      volumes:
        - name: google-cloud-creds
          secret:
            secretName: google-cloud-creds
```

**Comment by DwarfenM** [comment]
As workaround you can try to build you docker image with -Dvertx.disableFileCPResolving=true property.
For jvm build change this line:
`ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"`
To: 
`ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Dvertx.disableFileCPResolving=true -Djava.util.logging.manager=org.jboss.logmanager.LogManager"`

**Comment by cescoffier** [comment]
Yes, the workaround would work.

It's also fixed in #3016.

**Comment by kameshsampath** [comment]
Thanks @cescoffier

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
